test/TestUtilities/Test.Utility/Signing/SigningTestUtility.cs

```diff
@@ -678,18 +678,28 @@ public static void AssertOfflineRevocation(IEnumerable<ILogMessage> issues, LogL
         //We will change the original X509ChainStatus.StatusInformation of OfflineRevocation to VerifyCertTrustOfflineWhileRevocationModeOffline or VerifyCertTrustOfflineWhileRevocationModeOnline in Signature.cs and Timestamp.cs
         //So if we use APIs above to verify the results of chain.build, we should use assert AssertOfflineRevocationOnlineMode and AssertOfflineRevocationOfflineMode
         public static void AssertOfflineRevocationOnlineMode(IEnumerable<SignatureLog> issues, LogLevel logLevel)
+        {
+            AssertOfflineRevocationOnlineMode(issues, logLevel, NuGetLogCode.NU3018);
+        }
+
+        public static void AssertOfflineRevocationOnlineMode(IEnumerable<SignatureLog> issues, LogLevel logLevel, NuGetLogCode code)
         {
             Assert.Contains(issues, issue =>
-                issue.Code == NuGetLogCode.NU3018 &&
+                issue.Code == code &&
                 issue.Level == logLevel &&
                 issue.Message.Contains("The revocation function was unable to check revocation because the revocation server could not be reached. For more information, visit https://aka.ms/certificateRevocationMode."));
         }
 
         public static void AssertOfflineRevocationOfflineMode(IEnumerable<SignatureLog> issues)
+        {
+            AssertOfflineRevocationOfflineMode(issues, LogLevel.Information, NuGetLogCode.Undefined);
+        }
+
+        public static void AssertOfflineRevocationOfflineMode(IEnumerable<SignatureLog> issues, LogLevel logLevel, NuGetLogCode code)
         {
             Assert.Contains(issues, issue =>
-                issue.Code == NuGetLogCode.Undefined &&
-                issue.Level == LogLevel.Information &&
+                issue.Code == code &&
+                issue.Level == logLevel &&
                 issue.Message.Contains("The revocation function was unable to check revocation because the certificate is not available in the cached certificate revocation list and NUGET_CERT_REVOCATION_MODE environment variable has been set to offline. For more information, visit https://aka.ms/certificateRevocationMode."));
```

Comment:
Consider adding `[assembly: InternalsVisibleTo("Test.Utility")]` in [NuGet.Packaging\Properties\AssemblyInfo.cs](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Packaging/Properties/AssemblyInfo.cs) to avoid hardcoding the string message. That way, you can replace this line with:

```suggestion
                issue.Message.Contains(NuGet.Packaging.Strings.VerifyCertTrustOfflineWhileRevocationModeOffline));
```

Reply:
Thanks Fernando! That's cool! Good to know about it:)
I have one question, in AssemblyInfo.cs, for` #if SIGNED_BUILD` part, may I know how to find the value for PublicKey of Test.Utility? Or, do we need that? Thanks!

Reply:
Never mind. I think I've found it :)

Reply:
Fixed.